### PR TITLE
Fix New Relic agent configuration

### LINKED ISSUE #1792: Validate New Relic in ingesting data from Tock
## Description:

During the Tock Weekly Log Audit, we observed that no alerts have been generated since March of this year. This issue aims to:

image

Confirm the absence of alerts since March.
Ensure that New Relic is correctly ingesting data from Tock.

## Tasks:

- [ ] Validate the absence of alerts since March.
- [ ] Verify that New Relic is properly ingesting data from Tock.

### PR DESCRIPTION
**Pull request by nateborr**
## Description

Make two configuration changes in the New Relic agent configuration:

- Configure the Python agent to send data to `gov-collector.newrelic.com` to ensure FedRAMP compliance.
- Add the `gov-collector.newrelic.com` endpoint to Tock's egress proxy allow list, which should enable the Python agent to report APM data to New Relic again.

We are recording Deployment events successfully because Tock sends them via the `newrelic_admin` script. It sends data to `api.newrelic.com`, which we've already allowed through the egress proxy and which doesn't require a FedRAMP-specific setting.

Resolves #1792.

## Testing

The change in system behavior can only be confirmed in a deployed environment with an egress proxy.

Review that the egress proxy configuration change is unlikely to do harm and that the automated testing and checks pass.

After merge and deployment to staging, @nateborr will confirm whether New Relic is successfully receiving APM data.

## Additional information

* https://docs.newrelic.com/docs/new-relic-solutions/get-started/networks/#new-relic-endpoints
* https://docs.newrelic.com/docs/security/security-privacy/compliance/fedramp-compliant-endpoints/#agents
* https://docs.newrelic.com/docs/apm/agents/python-agent/installation/python-agent-admin-script-advanced-usage/#record-deploy
* https://github.com/18F/tock/pull/1734

### CODE REVIEW

File: egress_proxy/tock.vars.yml [hunk 1]
```diff
@@ -8,4 +8,5 @@ proxyallow: |
   uaa.fr.cloud.gov
   google-analytics.com
   api.newrelic.com
+  collector.newrelic.com
```
Comment (cantsin) [inline egress_proxy/tock.vars.yml:11]: So for my own understanding, `newrelic-admin` calls out to api.newrelic.com but NR proper uses `collector.newrelic.com`? 

Can we double check that these aren't using FedRamp'd endpoints -- e.g., `gov-collector.newrelic.com`?

Comment (nateborr) [reply egress_proxy/tock.vars.yml:11]: That's a good catch on the FedRamp collector endpoint. I'll confirm whether we're currently configured to send data there.

> So for my own understanding, newrelic-admin calls out to api.newrelic.com but NR proper uses collector.newrelic.com?

That's correct. New Relic's agents send APM data to the collector endpoint. The `newrelic-admin` script's record-deploy command [wraps the New Relic REST API](https://docs.newrelic.com/docs/apm/agents/python-agent/installation/python-agent-admin-script-advanced-usage/#record-deploy) at api.newrelic.com. As far as I can tell, there's not a separate FedRAMP-compliant endpoint for those API calls.

**Comment by nateborr** [comment]
@cantsin I've updated these changes to specify the `gov-collector.newrelic.com` endpoint, which we weren't doing previously. I'll update the PR description with some more detail.

I'm still waiting on confirmation from our New Relic admins that the Tock account in NR is configured for FedRAMP data storage. I'd like to confirm that before proceeding next week, but this should be safe to try out in staging in any case.

**Review by cantsin — APPROVED** [review]
LGTM -- many thanks for looking into this!

**Comment by cantsin** [comment]
Note that we'll have to do a manual push of the new proxy configuration in staging.

**Review by cantsin — APPROVED** [review]

**Pull request merged**